packages/syft/src/syft/core/node/new/task/task.py

```diff
@@ -0,0 +1,78 @@
+# stdlib
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+# third party
+from pydantic import BaseModel
+
+# relative
+from .....core.node.common.node_table.syft_object import SYFT_OBJECT_VERSION_1
+from .....core.node.common.node_table.syft_object import SyftObject
+from ....common.serde.serializable import serializable
+from ....common.uid import UID
+from ...abstract.node import AbstractNodeClient
+
+
+@serializable(recursive_serde=True)
+class NodeView(BaseModel):
+    class Config:
+        arbitrary_types_allowed = True
+
+    name: str
+    node_uid: UID
+
+    @staticmethod
+    def from_client(client: AbstractNodeClient):
+        return NodeView(name=client.name, node_uid=client.node_uid)
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, NodeView):
+            return False
+        elif self.name == other.name and self.node_uid == other.node_uid:
```

Comment:
MAYBE: Remove the `elif` and simply do return

Reply:
Done in [b1c9af5](https://github.com/OpenMined/PySyft/pull/7289/commits/b1c9af5f30c99117d59aa92dc15e8c9698d49b96)